### PR TITLE
Tracing console output with proper monotonic time

### PR DESCRIPTION
**Pull request by akoptelov**
- [x] Use standard tracing subscriber for console output
- [x] Specify monotonic time for trace events

Issue #44

### CODE REVIEW

**Comment by akoptelov** [comment]
@binier I've also added submodules, now it is easier to just modify our dependencies right in the project tree.

File: cli/src/commands/snarker/mod.rs [hunk 1]
```diff
@@ -34,23 +36,33 @@ mod http_server;
...
 }
 
 impl Snarker {
     pub fn run(self) -> Result<(), crate::CommandError> {
-        // stderrlog::new()
-        //     .verbosity(5)
-        //     // .module(module_path!()) // remove this line to see libp2p logs.
-        //     .init()
-        //     .unwrap();
+        unsafe { backtrace_on_stack_overflow::enable() };
```
Comment (binier) [inline cli/src/commands/snarker/mod.rs:63]: We don't need this anymore.

File: node/shared/src/block.rs [hunk 2]
```diff
@@ -10,18 +10,36 @@ pub use mina_p2p_messages::v2::StateHash as BlockHash;
 pub type ArcBlock = Arc<Block>;
 pub type ArcBlockWithHash = BlockWithHash<Arc<Block>>;
 
-#[derive(Serialize, Deserialize, Debug, Clone)]
+#[derive(Serialize, Deserialize, Clone)]
```
Comment (binier) [inline node/shared/src/block.rs:13]: Tmp changes that were in original branch, We should undo changes in this file.

File: cli/src/commands/snarker/http_server.rs [hunk 3]
```diff
@@ -69,7 +69,8 @@ pub async fn run(port: u16, rpc_sender: super::RpcSender) {
             let result: Option<RpcStateGetResponse> =
                 rpc_sender_clone.oneshot_request(RpcRequest::GetState).await;
 
-            with_json_reply(&result, StatusCode::OK)
+            let result = result.map(|v| v.transition_frontier.sync);
```
Comment (binier) [inline cli/src/commands/snarker/http_server.rs:72]: Tmp change, needs to be removed.

File: p2p/src/channels/rpc/mod.rs [hunk 4]
```diff
@@ -222,7 +222,7 @@ impl P2pRpcResponse {
 
         if matches!(result_kind, RpcResultKind::Err) {
             let err = mina_p2p_messages::rpc_kernel::Error::binprot_read(r)?;
-            let err = format!("{:?}", err);
+            let err = dbg!(format!("{:?}", err));
```
Comment (binier) [inline p2p/src/channels/rpc/mod.rs:225]: Tmp change, needs to be removed.

File: snarker/src/ledger/ledger_service.rs [hunk 5]
```diff
@@ -186,6 +186,12 @@ impl<T: LedgerService> TransitionFrontierService for T {
         block: ArcBlockWithHash,
         pred_block: ArcBlockWithHash,
     ) -> Result<(), String> {
+        eprintln!(
```
Comment (binier) [inline snarker/src/ledger/ledger_service.rs:189]: Tmp change, needs to be removed.

**Review by binier — COMMENTED** [review]
Looks good. We just need to revert extra changes caused by force push.

File: cli/src/commands/mod.rs [hunk 6]
```diff
@@ -3,10 +3,17 @@ pub mod snarker;
 
 pub type CommandError = Box<dyn std::error::Error>;
 
-#[derive(Debug, structopt::StructOpt)]
-#[structopt(name = "openmina", about = "Openmina Cli")]
+#[derive(Debug, clap::Parser)]
+#[command(name = "openmina", about = "Openmina Cli")]
+pub struct OpenminaCli {
```
[the remark this answers is not in the document]
Comment (akoptelov) [reply cli/src/commands/mod.rs:8]: This is how they suggest using `clap`. I was able to get rid of of that top-level struct, having the enum to be a `clap::Parser`, but it does not work for inner subcommands. Another reason, is that there might be some common options, and they should be added to the `OpenminaCli` struct. I would keep it like this.

File: cli/src/commands/snark/mod.rs [hunk 7]
```diff
@@ -1,16 +1,21 @@
 pub mod precalculate_block_verifier_index_and_srs;
 pub use precalculate_block_verifier_index_and_srs::PrecalculateBlockVerifierIndexAndSrs;
 
-#[derive(Debug, structopt::StructOpt)]
-#[structopt(name = "openmina", about = "Openmina Cli")]
-pub enum Snark {
+#[derive(Debug, clap::Args)]
+pub struct Snark {
```
[the remark this answers is not in the document]
Comment (akoptelov) [reply cli/src/commands/snark/mod.rs:5]: See my prev comment.

File: cli/src/commands/mod.rs (hunk 6 above)
[the remark this answers is not in the document]
Comment (binier) [reply cli/src/commands/mod.rs:8]: Yeah but from my experience, cli that has multiple layers on which args can be passed, is harder to work with. But ok, that works.

**Review by binier — APPROVED** [review]
Looks good, thanks.

**Pull request merged**